src/ng/compile.js

```diff
@@ -2540,9 +2545,23 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         lastValue,
         parentGet, parentSet, compare;
 
+        if ((attrName in attrs) && !attrs.hasOwnProperty(attrName)) {
+          // User defined a binding with same as a method in Object.prototype and also the attribute is not present
+          // Override property to make sure subsequent code do not have access to prototype method
+          attrs[attrName] = undefined;
+        }
+
         switch (mode) {
 
           case '@':
+            if (!attrs[attrName]) {
```

Comment:
As from line 2548
```javascript
if ((attrName in attrs) && !attrs.hasOwnProperty(attrName)) {
     // User defined a binding with same as a method in Object.prototype and also the attribute is not present
    // Override property to make sure subsequent code do not have access to prototype method
    attrs[attrName] = undefined;
}
```
has set `attrs[attrName]` to `undefined` in the case `attrName` is a method name but not actually defined in attributes. It is unnecessary to check with `hasOwnProperty `.